i3lock.c

```diff
@@ -1225,36 +1369,81 @@ static cairo_surface_t *read_raw_image(const char *image_path, const char *image
     return img;
 }
 
-static bool verify_png_image(const char *image_path) {
-    if (!image_path) {
-        return false;
-    }
-
-    /* Check file exists and has correct PNG header */
-    FILE *png_file = fopen(image_path, "r");
-    if (png_file == NULL) {
-        DEBUG("Image file path \"%s\" cannot be opened: %s\n", image_path, strerror(errno));
-        return false;
-    }
+static bool verify_png_image(FILE *png_file) {
     unsigned char png_header[8];
     memset(png_header, '\0', sizeof(png_header));
+    fseek(png_file, 0, SEEK_SET);
     int bytes_read = fread(png_header, 1, sizeof(png_header), png_file);
-    fclose(png_file);
     if (bytes_read != sizeof(png_header)) {
-        DEBUG("Could not read PNG header from \"%s\"\n", image_path);
+        DEBUG("Could not read PNG header\n");
         return false;
     }
 
     // Check PNG header according to the specification, available at:
     // https://www.w3.org/TR/2003/REC-PNG-20031110/#5PNG-file-signature
     static unsigned char PNG_REFERENCE_HEADER[8] = {137, 80, 78, 71, 13, 10, 26, 10};
     if (memcmp(PNG_REFERENCE_HEADER, png_header, sizeof(png_header)) != 0) {
-        DEBUG("File \"%s\" does not start with a PNG header. i3lock currently only supports loading PNG files.\n", image_path);
         return false;
     }
+
     return true;
 }
 
+static bool verify_gif_image(FILE *gif_file) {
+    unsigned char gif_header[6] = {0};
+    fseek(gif_file, 0, SEEK_SET);
+    int bytes_read = fread(gif_header, 1, sizeof(gif_header), gif_file);
+    if (bytes_read != sizeof(gif_header)) {
+        fprintf(stderr, "Could not read GIF header\n");
+        return false;
+    }
+
+    // Check GIF header according to the specification, available at:
+    // https://www.w3.org/Graphics/GIF/spec-gif89a.txt
+    static const unsigned char GIF_REFERENCE_HEADER[] = {'G', 'I', 'F'};
+    if (memcmp(GIF_REFERENCE_HEADER, gif_header, sizeof(GIF_REFERENCE_HEADER)) == 0) {
+        if (gif_header[3] >= '0' && gif_header[3] <= '9' &&
+            gif_header[4] >= '0' && gif_header[4] <= '9' &&
+            gif_header[5] >= 'a' && gif_header[5] <= 'z') {
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static enum IMAGE_FORMAT verify_image(const char *image_path) {
+    if (!image_path) {
+        return false;
+    }
+
+    if (image_raw_format != NULL) {
+        return IMAGE_FORMAT_RAW;
+    }
+
+    /* Check file exists and has correct header */
+    FILE *file = fopen(image_path, "r");
```

Comment:
Any specific reason `rb` from the original was switched to `r`?

Reply:
No, but it would be better to use `rb` for portability. Thanks!